bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/data/HueSettings.java

```diff
@@ -223,6 +223,10 @@ protected int count() {
 			return dataMap.size();
 		}
 
+		protected Set<String> getKeys(){
+			return dataMap.keySet();
+		}
```

Comment:
are there any other Key types then Integer? If not i would suggest to transform the Strings into Integer once (probably here?)